Makefile

```diff
@@ -158,53 +153,36 @@ LIBC_TOP_HALF_MUSL_SOURCES = \
                  %/copysignf.c %/copysign.c \
                  %/fminf.c %/fmaxf.c \
                  %/fmin.c %/fmax.c, \
-                 $(wildcard $(LIBC_TOP_HALF_MUSL_SRC_DIR)/math/*.c)) \
-    $(filter-out %/crealf.c %/creal.c \
-                 %/cimagf.c %/cimag.c, \
-                 $(wildcard $(LIBC_TOP_HALF_MUSL_SRC_DIR)/complex/*.c)) \
-    $(wildcard $(LIBC_TOP_HALF_MUSL_SRC_DIR)/crypt/*.c)
-MUSL_PRINTSCAN_SOURCES = \
-    $(LIBC_TOP_HALF_MUSL_SRC_DIR)/internal/floatscan.c \
-    $(LIBC_TOP_HALF_MUSL_SRC_DIR)/stdio/vfprintf.c \
-    $(LIBC_TOP_HALF_MUSL_SRC_DIR)/stdio/vfwprintf.c \
-    $(LIBC_TOP_HALF_MUSL_SRC_DIR)/stdio/vfscanf.c \
-    $(LIBC_TOP_HALF_MUSL_SRC_DIR)/stdlib/strtod.c \
-    $(LIBC_TOP_HALF_MUSL_SRC_DIR)/stdlib/wcstod.c
-LIBC_TOP_HALF_HEADERS_PRIVATE = $(LIBC_TOP_HALF_DIR)/headers/private
-LIBC_TOP_HALF_SOURCES = $(LIBC_TOP_HALF_DIR)/sources
-LIBC_TOP_HALF_ALL_SOURCES = \
-    $(LIBC_TOP_HALF_MUSL_SOURCES) \
-    $(shell find $(LIBC_TOP_HALF_SOURCES) -name \*.c)
-
-# Set the target variables. Multiarch triples notably omit the vendor
-# field, which happens to be what we do for the main target triple too.
-TARGET_TRIPLE = wasm32-wasi
-MULTIARCH_TRIPLE = wasm32-wasi
-
-# These variables describe the locations of various files and
-# directories in the generated sysroot tree.
-SYSROOT_LIB = $(SYSROOT)/lib/$(MULTIARCH_TRIPLE)
-SYSROOT_INC = $(SYSROOT)/include
-SYSROOT_SHARE = $(SYSROOT)/share/$(MULTIARCH_TRIPLE)
-
-# Set the target.
-override WASM_CFLAGS += --target=$(TARGET_TRIPLE)
-# WebAssembly floating-point match doesn't trap.
-# TODO: Add -fno-signaling-nans when the compiler supports it.
-override WASM_CFLAGS += -fno-trapping-math
-
-# Configure support for threads.
-ifeq ($(THREAD_MODEL), single)
-override WASM_CFLAGS += -mthread-model single
+                 $(wildcard $(LIBC_TOP_HALF_MUSL_SRC_DIR)/math/*.c))
+ifeq ($(BUILD_LIBC_BOTTOM_HALF),no)
+override CRT_SOURCES = $(BASICS_CRT_SOURCES)
+else
+override CRT_SOURCES = $(LIBC_BOTTOM_HALF_CRT_SOURCES)
 endif
-ifeq ($(THREAD_MODEL), posix)
-override WASM_CFLAGS += -mthread-model posix -pthread
+ifeq ($(LTO),yes)
+# The following files define functions which are called by LLVM CodeGen,
+# which runs after LLVM LTO, so LTO libraries don't satisfy them. They're
+# also relatively uninteresting to LTO, as LLVM recognizes most of them
+# even without seeing their definitions.
+override LIBC_NONLTO_SOURCES = \
```

Comment:
@sunfishcode 
@sbc100 
this exclusion is no longer necessary after https://reviews.llvm.org/D71738, right?